### PR TITLE
Issue #14937: Migrated section 7.3 Where javadoc is used to chapter wise testing

### PR DESCRIPTION
**Pull request by Zopsss**
#14937

### CODE REVIEW

**Comment by Zopsss** [comment]
failing CI: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/26347/workflows/5ff737aa-dc9c-4c16-b9b7-d7870bd341a1/jobs/624739/steps

not sure what's causing it to fail

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired/InputJavadocMethodAndMissingJavadocMethod.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
+package com.google.checkstyle.test.chapter7javadoc.rule73wherejavadocrequired;
+
+/**
+ * The following is a bad tag.
+ * @mytag Hello
+ */
+public class InputJavadocMethodAndMissingJavadocMethod extends OverrideClass
+{
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired/InputJavadocMethodAndMissingJavadocMethod.java:8]: Used only single input for `JavadocMethod` & `MissingJavadocMethod` modules, as they had the same input file before. They also work almost in the same way. They both logs violations for missing javadoc, I dont why we have two different modules for it

File: src/main/resources/google_checks.xml [hunk 2]
```diff
@@ -336,7 +336,7 @@
       <property name="tokens" value="METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF, COMPACT_CTOR_DEF"/>
     </module>
     <module name="MissingJavadocMethod">
-      <property name="scope" value="public"/>
+      <property name="scope" value="protected"/>
```
Comment (Zopsss) [inline src/main/resources/google_checks.xml:339]: Changes made based on following:

[7.3 Where Javadoc is used](https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required)

> At the minimum, Javadoc is present for every public class, and every public or protected member of such a class, with a few exceptions noted below.

We need to log violations for protected members too.

**Review by Zopsss — COMMENTED** [review]
Some important changes made:

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired/InputJavadocMethodAndMissingJavadocMethod.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+    /**
+     * Ok, missing params tags in javadoc are allowed
+     */
+    public InputJavadocMethodAndMissingJavadocMethod(double a, double b) {
+        foo2();
+        foo91();
+        foo5();
+        foo82();
+    }
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired/InputJavadocMethodAndMissingJavadocMethod.java:171]: Newly added test cases

**Review by Zopsss — COMMENTED** [review]
Forgot to include in previous review

File: src/main/resources/google_checks.xml (hunk 2 above)
Comment (romani) [reply src/main/resources/google_checks.xml:339]: We should not mix user affecting changes with internal migration/refactoring.
Please postpone this update to next PR.

Comment (Zopsss) [reply src/main/resources/google_checks.xml]: reverted this and also removed test cases related to this

**Comment by Zopsss** [comment]
failing CI: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/26352/workflows/4b58c4c7-eb5b-4c22-9808-6891d0400c3a/jobs/624957

maybe CircleCI had some internal issues or something?

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by romani** [comment]
Strange failure https://app.circleci.com/pipelines/github/checkstyle/checkstyle/26352/workflows/5ae6019a-97b9-4eb5-ba84-89296d49ffbb/jobs/624975/parallel-runs/0/steps/0-103

Restarted.

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/blob/3c4dcfb64ff87477aa2bcc52261bc25b63904ae6/.ci/validation.sh#L462

We need to update this shell function to use file that produces warning

**Review by romani — CHANGES_REQUESTED** [review]
See above

**Comment by Zopsss** [comment]
done @romani

**Review by romani — APPROVED** [review]
Ok to merge

**Review by rdiachenko — APPROVED** [review]
lgtm

**Comment by rdiachenko** [comment]
@Zopsss please resolve conflicts

**Comment by Zopsss** [comment]
@rdiachenko done

**Pull request merged**